scalyr_agent/monitor_utils/tests/k8s_test.py

```diff
@@ -232,3 +289,264 @@ def wait_for_requests(self, target_pending):
+        Only `obj` is used.  All other parameters are there to provide compatibility with the real
+        process_object
+
+        @param obj: a dict of key values that will be mapped to attributes of the result
+        """
+        return create_object_from_dict( obj )
+
+class FakeCache(object):
+    """Used in the test to fake out the KubernetesCache.
+
+    It allows for requests to the `pod` method to block until some other caller supplies what response
+    should be returned for it.
+    """
+    def __init__(self):
+
+        self.__processor = FakeProcessor()
+        self.__pod_cache = _K8sCache( self.__processor, 'Pod', perform_full_updates=False )
+        self.wait_timeout = 5
+        self.k8s = FakeK8s( wait_timeout=self.wait_timeout )
+        self.__clock = FakeClock()
+
+    def is_pod_cached(self, pod_namespace, pod_name, allow_expired):
+        """Faked KubernetesCache method that returns if the pod has been warmed from the cache's perspective.
+        @param pod_namespace: The namespace for the pod
+        @param pod_name:  The name for the pod
+        @param allow_expired: If True, an object is considered present in cache even if it is expired.
+
+        @type pod_namespace: str
+        @type pod_name: str
+        @type allow_expired: bool
+
+        @return True if the pod has been warmed.
+        @rtype: bool
+        """
+        return self.__pod_cache.is_cached( pod_namespace, pod_name, allow_expired )
+
+    def pod(self, pod_namespace, pod_name, allow_expired=False, current_time=None, query_options=None ):
+        """Faked KubernetesCache method that simulates blocking for the specified pod's cached entry.
+
+        @param pod_namespace: The namespace for the pod
+        @param pod_name:  The name for the pod
+
+        @type pod_namespace: str
+        @type pod_name: str
+        """
+        return self.__pod_cache.lookup(self.k8s, current_time, pod_namespace, pod_name, kind='Pod',
+                                       allow_expired=allow_expired, query_options=query_options)
+
+    def stop(self):
+        """Stops the cache.  Called when the test is finished.
+        """
+        self.k8s.stop()
+
+    def set_response(self, namespace, name, **kwargs ):
+        self.k8s.set_response( namespace, name, **kwargs )
+
+    def wait_until_request_pending(self, namespace=None, name=None ):
+        return self.k8s.wait_until_request_pending( namespace=namespace, name=name )
+
+    def wait_until_request_finished(self, namespace, name):
+        return self.k8s.wait_until_request_finished( namespace, name )
+
+    def simulate_add_pod_to_cache( self, pod_namespace, pod_name ):
+        """
+        Simulates adding a pod to the cache so that we can populate the
+        cache for testing purposes without going through the regular interface
+        """
+        self.k8s.set_response( pod_namespace, pod_name, success=True )
```

Comment:
If you are adding the pod to the cache (as you do below), why do you need to set an API response for it?  If it is in the cache, the api should never be queried for it, correct?

Reply:
This was accidentally left over from a previous revision of the code.  I've now removed it.